### PR TITLE
feat: relottie-metadata plugin

### PR DESCRIPTION
**Pull request by Aidosmf**
## Description

Initial major release for the new relottie plugin. For more information please read its readme.md

## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [x]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] This is something we need to do.

### CODE REVIEW

File: packages/relottie-metadata/__tests__/__snapshots__/extra/sale.snap [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`relottie-metadata fixtures extra Test in sale.json should have correct data 1`] = `
+{
+  "colors": [
+    {
+      "hex": "none",
+      "nodeTitle": "theme-color",
+      "rgba": [
+        NaN,
```
Comment (Aidosmf) [inline packages/relottie-metadata/__tests__/__snapshots__/extra/sale.snap:10]: hi

File: packages/relottie-metadata/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+type FileData = ParseFileData & StringifyFileData & MetadataFileData
+
+// The Lottie source: __fixtures__/features/auto-orient.json
+const lottieJson = '...';
+
+const vfile = unified()
+  .use(relottieParse)
+  .use(relottieMetadata)
+  .use(relottieStringify)
```
Comment (theashraf) [inline packages/relottie-metadata/README.md:29]: [Question]
the `relottieStringify` plugin is optional in this use case i assume ?

File: packages/relottie-metadata/__tests__/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    const FEATURES_DIR_NAME = 'features';
+    const EXTRA_DIR_NAME = 'extra';
+
+    describe(FEATURES_DIR_NAME, () => {
+      const dirPath = `${FIXTURE_DIR_PATH + FEATURES_DIR_NAME}/`;
+      const snapshotDirPath = `${SNAPSHOT_DIR_PATH + FEATURES_DIR_NAME}/`;
+
+      const files = readdirSync(dirPath);
+
+      files.forEach((file) => {
```
Comment (theashraf) [inline packages/relottie-metadata/__tests__/index.test.ts:38]: [Suggestion]
you may consider using `test.each`
https://jestjs.io/docs/api#testeachtablename-fn-timeout

File: packages/relottie-metadata/src/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    const hexToken = Math.round(val * 255).toString(16);
+
+    hexTokens.push(hexToken.length === 1 ? `0${hexToken}` : hexToken);
+  }
+
+  return `#${hexTokens.join('')}`;
+}
+
+export const isHexValid = (value: string): boolean => /^#[\da-f]{6}$/iu.test(value);
```
Comment (theashraf) [inline packages/relottie-metadata/src/helpers.ts:88]: Shouldn't it be an 8-digit hex code to include the alpha component of the color? I believe Lottie uses color codes that include the alpha component, so an RRGGBBAA format is expected here

Comment (theashraf) [reply packages/relottie-metadata/src/helpers.ts:88]: there are several valid hex color formats:

Short form:
RGB (e.g., #FFF)
RGBA (e.g., #FFFF)

Long form:
RRGGBB (e.g., #FFFFFF)
RRGGBBAA (e.g., #FFFFFFFF)

File: packages/relottie-metadata/README.md (hunk 2 above)
Comment (Aidosmf) [reply packages/relottie-metadata/README.md:29]: you cannot call `.process` or `.processSync` without a Compiler

File: packages/relottie-metadata/src/helpers.ts (hunk 4 above)
Comment (Aidosmf) [reply packages/relottie-metadata/src/helpers.ts]: fixed and pushed, thanks

**Review by theashraf — APPROVED** [review]

**Pull request merged**